cypress/integration/Homework/calculator.spec.js

```diff
@@ -0,0 +1,408 @@
+                calculator.getSecondNumberField().type(secondSymbol)
+                calculator.getOperationDropDownList().select('Add')
+                calculator.getCalculateButton().click()
+                calculator.getErrorField().contains(errorMessage)
+                calculator.getAnswerField().should('be.empty')
+            })
+            
+            it("TS5.2: Performs the operation 'Subtract' with symbols " + firstSymbol + " and " + secondSymbol + " (Integers only set to - " + integersOnly + " ) and returns the following error message - " + errorMessage, () => {
+                calculator.getFirstNumberField().clear()
+                calculator.getSecondNumberField().clear()
+                calculator.getFirstNumberField().type(firstSymbol)
+                calculator.getSecondNumberField().type(secondSymbol)
+                calculator.getOperationDropDownList().select('Subtract')
+                calculator.getCalculateButton().click()
+                calculator.getErrorField().contains(errorMessage)
+                calculator.getAnswerField().should('be.empty')
+            })
+            
+            it("TS5.3: Performs the operation 'Multiply' with symbols " + firstSymbol + " and " + secondSymbol + " (Integers only set to - " + integersOnly + " ) and returns the following error message - " + errorMessage, () => {
+                calculator.getFirstNumberField().clear()
+                calculator.getSecondNumberField().clear()
+                calculator.getFirstNumberField().type(firstSymbol)
+                calculator.getSecondNumberField().type(secondSymbol)
+                calculator.getOperationDropDownList().select('Multiply')
+                calculator.getCalculateButton().click()
+                calculator.getErrorField().contains(errorMessage)
+                calculator.getAnswerField().should('be.empty')
+            })
+            
+            it("TS5.4: Performs the operation 'Divide' with symbols " + firstSymbol + " and " + secondSymbol + " (Integers only set to - " + integersOnly + " ) and returns the following error message - " + errorMessage, () => {
+                calculator.getFirstNumberField().clear()
+                calculator.getSecondNumberField().clear()
+                calculator.getFirstNumberField().type(firstSymbol)
+                calculator.getSecondNumberField().type(secondSymbol)
+                calculator.getOperationDropDownList().select('Divide')
+                calculator.getCalculateButton().click()
+                calculator.getErrorField().contains(errorMessage)
+                calculator.getAnswerField().should('be.empty')
+            })
+        })
+        
+        describe('TS6 (Build - ' + buildVersion + '): Calculator "Clear" button functionality testing with numbers', () => {
+            
+            before("Visits the initial page", () => {
+                calculator.visitInitialPage()
```

Comment:
I agree with this section. This could be extracted to a single point before all of the 'describe' sections. One question: should the website be visited before each test or only before each 'describe' section?

Reply:
It depends what you want to have - think about speed and reliability, one test failure should not affect other tests. Try out.